src/common/com/intellij/plugins/haxe/model/HaxeDocumentModel.java

```
@@ -83,12 +92,25 @@ public void wrapElement(final PsiElement element, final String before, final Str
   public void addTextBeforeElement(final PsiElement element, final String text) {
     if (element == null) return;
     TextRange range = element.getTextRange();
-    document.replaceString(range.getStartOffset(), range.getStartOffset(), text);
+    int start = range.getStartOffset();
+    document.replaceString(start, start, text);
+    reformat(start, start + text.length());
   }
 
   public void addTextAfterElement(final PsiElement element, final String text) {
     if (element == null) return;
```

This should throw or return an error, not just silently fail.